src/org/zaproxy/zap/extension/simpleExample/ExtensionSimpleExample.java

```diff
@@ -45,147 +46,151 @@
  */
 public class ExtensionSimpleExample extends ExtensionAdaptor {
 
-	// The name is public so that other extensions can access it
-	public static final String NAME = "ExtensionSimpleExample";
-	
-	// The i18n prefix, by default the package name - defined in one place to make it easier
-	// to copy and change this example
-	protected static final String PREFIX = "simpleExample";
+    // The name is public so that other extensions can access it
+    public static final String NAME = "ExtensionSimpleExample";
+
+    // The i18n prefix, by default the package name - defined in one place to make it easier
+    // to copy and change this example
+    protected static final String PREFIX = "simpleExample";
+
+    private static final String RESOURCE = "/org/zaproxy/zap/extension/simpleExample/resources";
+
+    private static final ImageIcon ICON = new ImageIcon(
+            ExtensionSimpleExample.class.getResource( RESOURCE + "/cake.png"));
+
+    private static final String EXAMPLE_FILE = "example/ExampleFile.txt";
 
-	private static final String RESOURCE = "/org/zaproxy/zap/extension/simpleExample/resources";
-	
-	private static final ImageIcon ICON = new ImageIcon(
-			ExtensionSimpleExample.class.getResource( RESOURCE + "/cake.png"));
+    private ZapMenuItem menuExample;
+    private RightClickMsgMenu popupMsgMenuExample;
+    private AbstractPanel statusPanel;
 
-	private static final String EXAMPLE_FILE = "example/ExampleFile.txt";
-	
-	private ZapMenuItem menuExample;
-	private RightClickMsgMenu popupMsgMenuExample;
-	private AbstractPanel statusPanel;
+    private SimpleExampleAPI api = new SimpleExampleAPI(this);

```

Comment:
The initialisation can be delayed, to `init()` (or `hook`).

Reply:
Shouldn't this be the same for the `LOGGER` variable as well then?

Reply:
That one is final so it must be initialised earlier and being a class variable it depends on how/where it's being used (though in this example it's not heavily used). The extension might be disabled so initialising the API does not add much (just creates unnecessary garbage at some point).